src/main/scala/com/fulcrumgenomics/bam/Bams.scala

```diff
@@ -145,16 +149,34 @@ object Bams extends LazyLogging {
     * @return an Iterator with reads from the same query grouped together
     */
   def queryGroupedIterator(in: SamReader,
-                           maxInMemory: Int = 1e6.toInt,
+                           maxInMemory: Int = MaxInMemory,
                            tmpDir: Option[DirPath] = None): BetterBufferedIterator[SAMRecord] = {
-    if (in.getFileHeader.getSortOrder == SortOrder.queryname || in.getFileHeader.getGroupOrder == GroupOrder.query) {
-      in.iterator().bufferBetter
+    queryGroupedIterator(in.iterator(), in.getFileHeader, maxInMemory, tmpDir)
+  }
+
+  /**
+    * Returns an iterator over the records in the given reader in such a way that all
+    * reads with the same query name are adjacent in the iterator. Does NOT guarantee
+    * a queryname sort, merely a query grouping.
+    *
+    * @param iterator an iterator from which to consume records
+    * @param header the header associated with the records.
+    * @param maxInMemory the maximum number of records to keep and sort in memory if sorting is needed
+    * @param tmpDir an optional temp directory to use for temporary sorting files if needed
+    * @return an Iterator with reads from the same query grouped together
+    */
+  def queryGroupedIterator(iterator: Iterator[SAMRecord],
+                           header: SAMFileHeader,
+                           maxInMemory: Int,
+                           tmpDir: Option[DirPath]): BetterBufferedIterator[SAMRecord] = {
+    if (header.getSortOrder == SortOrder.queryname || header.getGroupOrder == GroupOrder.query) {
+      iterator.bufferBetter
```

Comment:
I opted to have the same name rather than using something like the [Magnet pattern](http://spray.io/blog/2012-12-13-the-magnet-pattern/).